### PR TITLE
Logical operators

### PR DESCRIPTION
**Pull request by Mr-Saxobeat**
Translate article, tasks and solutions.

### CODE REVIEW

**Comment by Mr-Saxobeat** [comment]
> Error: the article already has PR number in the Progress Issue #1, it's 55 ⁉️

@iliakan What I should do? I've already closed the #55 because I forgot to translate the tasks that before.

**Comment by Mr-Saxobeat** [comment]
I'm sorry, I didn't know that I should commit to the older pull request #55 .

File: 1-js/02-first-steps/11-logical-operators/7-check-if-out-range/task.md [hunk 1]
```diff
@@ -1,9 +1,9 @@
...
 ---
 
-# Check the range outside
+# Verifique o exterior do intervalo
 
-Write an `if` condition to check that `age` is NOT between 14 and 90 inclusively.
+Escreva uma condição `if` para verificar se `idade` NÃO está entre o intervalo fechado `14` e `90`.
 
-Create two variants: the first one using NOT `!`, the second one -- without it.
+Crie duas variações: a primeira usando NÃO `!`, e uma segunda -- sem ele.
```
Comment (odsantos) [inline 1-js/02-first-steps/11-logical-operators/7-check-if-out-range/task.md]: - "Crie duas <s>variações</s> variantes: ..."

File: 1-js/02-first-steps/11-logical-operators/article.md [hunk 2]
```diff
@@ -27,83 +27,83 @@ alert( true || false );  // true
...
+1. **Obtendo o primeiro valor verdadeiro de uma lista de variáveis ou expressões.**
 
-    Imagine we have several variables which can either contain data or be `null/undefined`. How can we find the first one with data?
+    Imagine que temos várias variáveis que podem conter algum dado ou ser `null/undefined`. Como podemos encontrar a primeira com algum dado?
 
-    We can use OR `||`:
+    Nós podemos usar OU `||`:
 
     ```js run
     let currentUser = null;
```
Comment (odsantos) [inline 1-js/02-first-steps/11-logical-operators/article.md]: - "Nós podemos usar <s>OU</s> OR `||`:"

**Comment by odsantos** [comment]
Hi @Mr-Saxobeat you're still on this?
If so, could please also  resolve the conflicts?
Thank you.

**Comment by herbertpdl** [comment]
Hello @odsantos, I would like to collaborate. I saw that this topic is stopped from a long time.
Could I start for this one? 

Thank you!

**Comment by odsantos** [comment]
Hi @herbertpdl we'd appreciate it if you could translate Thank you.

**Comment by jonnathan-ls** [comment]
This PR has been open for a long time and there has been no progress on the requested review.

Before evaluating what will be done with this demand:

@Mr-Saxobeat: are you still interested in continuing the translation? If so, please check the notes recorded in the requested code review by @odsantos, after correction, I will carry out my review.

@herbertpdl: if MrSaxobeat, not interested or not responding, would you like to do a translation?

File: 1-js/02-first-steps/11-logical-operators/article.md [hunk 3]
```diff
@@ -1,24 +1,24 @@
-# Logical operators
+# Operadores lógicos
 
-There are three logical operators in JavaScript: `||` (OR), `&&` (AND), `!` (NOT).
+Existem três operadores lógicos em JavaScript: `||` (OU), `&&` (E), `!` (NÃO).
 
-Although they are called "logical", they can be applied to values of any type, not only boolean. Their result can also be of any type.
+Embora eles sejam chamados de "lógicos", podem ser aplicados a valores de qualquer tipo, não apenas a boolean. Seus resultados também podem ser de qualquer tipo.
```
Comment (jonnathan-ls) [inline 1-js/02-first-steps/11-logical-operators/article.md:5]:
```suggestion
Embora eles são denominados como "lógicos", podem ser aplicados a valores de qualquer tipo, não apenas ao tipo `boolean`. Seus resultados também podem ser de qualquer tipo.
```

File: 1-js/02-first-steps/11-logical-operators/article.md [hunk 4]
```diff
@@ -1,24 +1,24 @@
...
 
-The "OR" operator is represented with two vertical line symbols:
+O operador "OU" é representado com dois símbolos de linha vertical:
 
 ```js
 result = a || b;
 ```
 
-In classical programming, the logical OR is meant to manipulate boolean values only. If any of its arguments are `true`, it returns `true`, otherwise it returns `false`.
+Em programação clássica, o operador OU é mencionado para manipular apenas valores booleanos. Se qualquer um dos seus argumentos for `true`, ele retorna `true`, se não, retorna `false`.
```
Comment (jonnathan-ls) [inline 1-js/02-first-steps/11-logical-operators/article.md:17]: Complementando a sugestão do @odsantos em [", o operador lógico OU é <s>mencionado</s> utilizado para manipular ..."](https://github.com/javascript-tutorial/pt.javascript.info/pull/56/files#r358993776)

```suggestion
Em programação clássica, o operador OU é utilizado para manipular apenas valores do tipo `boolean`. Se qualquer um dos seus argumentos for `true`, ele retorna `true`, se não, retorna `false`.
```

File: 1-js/02-first-steps/11-logical-operators/article.md [hunk 5]
```diff
@@ -1,24 +1,24 @@
...
 
-In JavaScript, the operator is a little bit trickier and more powerful. But first, let's see what happens with boolean values.
+Em JavaScript, este operador é um pouco mais útil e poderoso. Mas primeiro, vamos ver o que acontece com valores booleanos.
```
Comment (jonnathan-ls) [inline 1-js/02-first-steps/11-logical-operators/article.md:19]:
```suggestion
Em JavaScript, este operador é um pouco mais útil e poderoso. Mas primeiro, vamos ver o que acontece com valores do tipo `boolean`.
```

File: 1-js/02-first-steps/11-logical-operators/article.md [hunk 6]
```diff
@@ -27,83 +27,83 @@ alert( true || false );  // true
 alert( false || false ); // false
 ```
 
-As we can see, the result is always `true` except for the case when both operands are `false`.
+Como podemos ver, o resultado é sempre `true`, exceto para o caso onde os dois operandos são `false`.
 
-If an operand is not a boolean, it's converted to a boolean for the evaluation.
+Se um operando não é um boolean, ele é convertido em um boolean para ser avaliado.
```
Comment (jonnathan-ls) [inline 1-js/02-first-steps/11-logical-operators/article.md:32]:
```suggestion
Se um operando não é um `boolean`, ele é convertido em um `boolean` para ser avaliado.
```

File: 1-js/02-first-steps/11-logical-operators/article.md [hunk 7]
```diff
@@ -27,83 +27,83 @@ alert( true || false );  // true
...
-- If all operands have been evaluated (i.e. all were `false`), returns the last operand.
+- Avalia os operandos da esquerda para a direita.
+- Para cada operando, o converte para boolean. Se o resultado é `true`, para e retorna o valor original daquele operando.
+- Se todos os operandos foram avaliados (i.e. todos são `false`), retorna o último operando.
 
-A value is returned in its original form, without the conversion.
+Um valor é retornado na sua forma original, sem conversão.
 
-In other words, a chain of OR `"||"` returns the first truthy value or the last one if no such value is found.
+Em outras palavras, uma cadeia de OU `"||"` retorna o primeiro valor verdadeiro ou o último se nenhum `true` for encontrado.
```
Comment (jonnathan-ls) [inline 1-js/02-first-steps/11-logical-operators/article.md:87]:
```suggestion
Em outras palavras, uma cadeia de OU `"||"`, retorna o primeiro valor verdadeiro ou o último se nenhum `true` for encontrado.
```

File: 1-js/02-first-steps/11-logical-operators/article.md [hunk 8]
```diff
@@ -113,27 +113,27 @@ This leads to some interesting usage compared to a "pure, classical, boolean-onl
...
+    alert( name ); // seleciona "John" - o primeiro valor verdadeiro
     ```
 
-    If both `currentUser` and `defaultUser` were falsy, `"unnamed"` would be the result.
-2. **Short-circuit evaluation.**
+    Se ambos `currentUser` e `defaultUser` forem falsos, o resultado será `"unnamed"`.
+2. **Avaliação de curto-circuito**
 
-    Operands can be not only values, but arbitrary expressions. OR evaluates and tests them from left to right. The evaluation stops when a truthy value is reached, and the value is returned. This process is called "a short-circuit evaluation" because it goes as short as possible from left to right.
+    Operandos podem não ser apenas valores, mas operações arbitrárias. OU interpreta e testa elas da esquerda para a direita. A avaliação é interrompida quando um valor verdadeiro é encontrado e este valor é retornado. Este processo é chamado de "avaliação de curto-circuito" pois vai o mais curto possível da esquerda para a direita.
```
[the remark this answers is not in the document]
Comment (jonnathan-ls) [reply 1-js/02-first-steps/11-logical-operators/article.md:122]: Complementando a sugestão

```suggestion
    Operandos podem não ser apenas valores, mas operações arbitrárias. `OR` interpreta e testa elas da esquerda para a direita. A avaliação é interrompida quando um valor verdadeiro é encontrado e este valor é retornado. Este processo é denominado como "avaliação em curto-circuito", pois vai o mais curto possível da esquerda para a direita.
```

File: 1-js/02-first-steps/11-logical-operators/article.md [hunk 9]
```diff
@@ -143,21 +143,21 @@ This leads to some interesting usage compared to a "pure, classical, boolean-onl
     alert(x); // 1
     ```
 
-    An assignment is a simple case. Other side effects can also be involved.
+    Uma atribuição é um caso simples. Outros ##SIDE EFFECTS(???)## podem também estarem envolvidos.
 
-    As we can see, such a use case is a "shorter way of doing `if`". The first operand is converted to boolean. If it's false, the second one is evaluated.
+    Como podemos ver, esse caso é como "uma maneira mais curta de se usar `if`". O primeiro operando é convertido para boolean. Se for false, o segundo operando é avaliado.
```
Comment (jonnathan-ls) [inline 1-js/02-first-steps/11-logical-operators/article.md:148]: Complementando a sugestão do @odsantos em [Dec 17, 2019](https://github.com/javascript-tutorial/pt.javascript.info/pull/56/files#r359009516)

```suggestion
    Como podemos ver, este caso é como "uma maneira mais curta de se usar `if`". O primeiro operando é convertido para o tipo `boolean`. Se for `false`, o segundo operando é avaliado.
```

**Review by jonnathan-ls — CHANGES_REQUESTED** [review]
Segue minha revisão!

**Comment by jonnathan-ls** [comment]
@odsantos adicionei minha revisão também.

Considerando que este PR e bastante antigo e como não tivemos retorno do @Mr-Saxobeat, pensei em utilizar a tradução realizada, corrigir as sugestões que solicitamos e realizar o `commit` diretamente neste PR, o que acha ? 

> Tudo bem realizarmos desta forma ?

**Comment by odsantos** [comment]
@jonnathan-ls I agree. Probably we could resolve the conflicts, revise each other, or have another maintainer joined to this PR, and have it merged when at least two maintainers see the translation ready for that.

File: 1-js/02-first-steps/11-logical-operators/article.md [hunk 10]
```diff
@@ -27,83 +27,83 @@ alert( true || false );  // true
 alert( false || false ); // false
 ```
```
[the remark this answers is not in the document]
Comment (jonnathan-ls) [reply 1-js/02-first-steps/11-logical-operators/article.md:29]: O trecho de código está correto @odsantos.

File: 1-js/02-first-steps/11-logical-operators/article.md [hunk 11]
```diff
@@ -143,21 +143,21 @@ This leads to some interesting usage compared to a "pure, classical, boolean-onl
     alert(x); // 1
     ```
```
[the remark this answers is not in the document]
Comment (jonnathan-ls) [reply 1-js/02-first-steps/11-logical-operators/article.md:145]: O trecho de código esta correto.

**Comment by jonnathan-ls** [comment]
> English

Considering that this PR is abandoned and the branch that the user used in the merge creation is no longer available so that it would be possible to contribute directly, a new PR #401 was opened replicating the existing translation in this PR with the corrections of the revision suggestions presented.

Therefore, we are closing this PR.

> Portuguese

Considerando que este PR está abandonado e a branch que o usuário utilizou na criação do merge não está mais disponível para que fosse possível contribuir diretamente, foi aberto um novo PR #401  replicando a tradução existente neste PR com as correções das sugestões de revisão apresentadas

Portanto, estamos dando como encerrado este PR.
